test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -0,0 +1,342 @@
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write events.
+        log.info("Writing event e1 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e2 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e3 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e4 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e5 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e6 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e7 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        readerGroupManager.createReaderGroup(READER_GROUP_1, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        readerGroupManager.createReaderGroup(READER_GROUP_2, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+
+        ReaderGroup readerGroup1 = readerGroupManager.getReaderGroup(READER_GROUP_1);
+        ReaderGroup readerGroup2 = readerGroupManager.getReaderGroup(READER_GROUP_2);
+        @Cleanup
+        EventStreamReader<String> reader1 = clientFactory.createReader(READER_GROUP_1 + "-" + 1,
+                READER_GROUP_1, new JavaSerializer<>(), readerConfig);
+        @Cleanup
+        EventStreamReader<String> reader2 = clientFactory.createReader(READER_GROUP_2 + "-" + 1,
+                READER_GROUP_2, new JavaSerializer<>(), readerConfig);
+
+        // Read three events with reader1.
+        log.info("Reader1 reading event e1 from {}/{}", SCOPE, STREAM);
+        EventRead<String> read = reader1.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read.getEvent());
+
+        log.info("Reader1 Reading event e2 from {}/{}", SCOPE, STREAM);
+        read = reader1.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read.getEvent());
+
+        log.info("Reader1 Reading event e3 from {}/{}", SCOPE, STREAM);
+        read = reader1.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read.getEvent());
```

Comment:
Can we use looping here to read the event?

Reply:
done